src/algebra/ordered_group.lean

```diff
@@ -1808,7 +1801,30 @@ abs_sub_le_iff.2 ⟨sub_le_sub hau hbl, sub_le_sub hbu hal⟩
 lemma eq_of_abs_sub_nonpos (h : abs (a - b) ≤ 0) : a = b :=
 eq_of_abs_sub_eq_zero (le_antisymm h (abs_nonneg (a - b)))
 
-end decidable_linear_ordered_add_comm_group
+lemma exists_gt_zero [nontrivial α] : ∃ (a:α), 0 < a :=
+begin
+  obtain ⟨y, hy⟩ := exists_ne (0 : α),
+  cases hy.lt_or_lt,
+  { exact ⟨- y, neg_pos.mpr h⟩ },
+  { exact ⟨y, h⟩ }
+end
+
+@[priority 100] -- see Note [lower instance priority]
+instance linear_ordered_add_comm_group.to_no_top_order [nontrivial α] :
+  no_top_order α :=
```

Comment:
Sorry, I broke the build by moving this instance from `linear_ordered_semiring` to `linear_ordered_add_comm_group`.  I had thought the nonexistence of `linear_ordered_add_comm_group` (as opposed to `linear_ordered_add_comm_group`) was the only thing stopping this lemma from existing in the right generality, but in fact the right generality is the nonexistent `linear_ordered_add_comm_monoid`.  I'll move it back.)

Reply:
`linear_ordered_add_comm_monoid` wouldn't work: you need a positive element. Counter-example: nonpositive integers.

Reply:
OK, right.  It's a shame, it's true both for `nontrivial` `add_comm_group` and for `semiring`, but I assume one can't have both, because then on `ring` there will be duplicate instances.

Reply:
That's not a problem for `Prop`-valued typeclasses.

Reply:
Oh!  Then I'll add the other version back in.  Thanks for explaining.